pkg/gardenlet/controller/seed/seed/components.go

```diff
@@ -284,14 +284,20 @@ func defaultSystem(
 		replicasExcessCapacityReservation = int32(numberOfZones)
 	}
 
+	coreSeed, err := helper.ConvertSeed(seed.GetInfo())
+	if err != nil {
+		return nil, err
+	}
+
 	return seedsystem.New(
 		c,
 		gardenNamespaceName,
 		seedsystem.Values{
 			ReserveExcessCapacity: seedsystem.ReserveExcessCapacityValues{
-				Enabled:  reserveExcessCapacity,
+				Enabled:  helper.SeedSettingExcessCapacityReservationEnabled(coreSeed.Spec.Settings),
```

Comment:
You could add `SeedSettingExcessCapacityReservationEnabled` to `pkg/apis/core/v1beta1/helper` (we usually have "duplication" for these functions) to spare conversion. I think it's even safe to move the helper because `SeedSettingExcessCapacityReservationEnabled` isn't used elsewhere.